src/DotVVM.Framework/Hosting/DotvvmPresenter.cs

```diff
@@ -249,6 +262,14 @@ public class DotvvmPresenter : IDotvvmPresenter
             }
         }
 
+        private long addTraceData(long lastStopwatchState, string processName, IDotvvmRequestContext context)
+        {
+            var stopwatch = context.Configuration.ServiceLocator.GetService<IStopwatch>();
```

Comment:
We should get the stopwatch instance in the constructor so we do not call `GetService` for every traced event.